python/lsst/obs/lsst/_ingestPhotodiode.py

```diff
@@ -143,34 +143,67 @@ def run(self, locations, run=None, file_filter=r".*Photodiode_Readings.*txt",
         numFailed = 0
         for inputFile in files:
             # Convert the file into the right class.
-            with inputFile.as_local() as localFile:
-                calib = PhotodiodeCalib.readTwoColumnPhotodiodeData(localFile.ospath)
+            # import pdb; pdb.set_trace()
+            calibType = "Unknown"
+            try:
+                # Can this be read directly?
+                with inputFile.as_local() as localFile:
+                    calib = PhotodiodeCalib.readText(inputFile.ospath)
+                calibType = "full"
+            except Exception:
+                # Try reading as a two-column file
+                with inputFile.as_local() as localFile:
+                    calib = PhotodiodeCalib.readTwoColumnPhotodiodeData(localFile.ospath)
+                calibType = "two-column"
+
+            # Get exposure records
+            if calibType == "full":
+                instrumentName = calib.getMetadata().get('INSTRUME')
+                if instrumentName is None:
+                    # The field is populated by the calib class, so we
+                    # can't use defaults.
+                    instrumentName = self.instrument.getName()
+
+                obsId = calib.getMetadata()['obsId']
+                whereClause = "exposure.obs_id=obsId"
+                exposureRecords = [rec for rec in registry.queryDimensionRecords("exposure",
+                                                                                 instrument=instrumentName,
+                                                                                 where=whereClause,
+                                                                                 bind={"obsId": obsId})]
+                idTuple = (obsId)
```

Comment:
```suggestion
                idTuple = (obsId,)
```
otherwise it's not a tuple and it looks like you want a tuple.

```python
>>> x = ("id")
>>> y = ("id",)
>>> type(x)
<class 'str'>
>>> type(y)
<class 'tuple'>
```
Given you only log the contents you probably don't need a tuple but then consider a different variable name.

Reply:
Changed to simply be `logId` (of `str` or `tuple` as needed), as this is just for the log messages.